### DISCUSSION TITLE
How to run without root on Fedora?

### DISCUSSION THREAD

**Discussion by srcid** [xremap/xremap#496] [Q&A]
I've changed to Fedora Linux, recently, and can't get xremap to work without root.

I tried this:

```
sudo gpasswd -a YOUR_USER input
echo 'KERNEL=="uinput", GROUP="input", TAG+="uaccess"' | sudo tee /etc/udev/rules.d/input.rules
```

and tried this too:
```
echo 'KERNEL=="event*", NAME="input/%k", MODE="660", GROUP="input"' | sudo tee /etc/udev/rules.d/input.rules
```

I got this eror:

```
Error: Failed to prepare an output device: Permission denied (os error 13)
```

The input group already exists in fedora. The devices in /dev/input do belong to input group, and so my user belongs to it. The evtest program does work without root. And I also change the selinux mode to permissive and disabled.

![Screenshot from 2024-06-17 20-09-33](https://github.com/xremap/xremap/assets/34487193/28341a42-b411-4b5b-9cf7-6074747da37a)

**Comment by alecbcs** [comment]
@srcid you might not be loading the `uinput` kernel module at boot. (Or at least that's what it was in my case when I ran into this error.) 
To fix it I ran the following from the Debian guide and rebooted my system.

```
echo uinput | sudo tee /etc/modules-load.d/uinput.conf
```

**Reply by srcid** [reply]
Thank you @alecbcs, it solves the problem. I didn't think that the module were not loaded.

**Comment by srcid** [comment]
The uinput module were not loaded at boot time. Just execute the command `echo uinput | sudo tee /etc/modules-load.d/uinput.conf` and GG.
